tracer/src/Datadog.Trace/RuntimeMetrics/PssRuntimeInformation.cs

```diff
@@ -0,0 +1,226 @@
+
+                if (result != 0)
+                {
+                    throw new Win32Exception(result, $"PssQuerySnapshot with PSS_QUERY_PROCESS_INFORMATION (64 bits) failed with code {result}");
+                }
+
+                userTime = processInformation.UserTime;
+                kernelTime = processInformation.KernelTime;
+                memoryUsage = (long)processInformation.PrivateUsage;
+            }
+            else
+            {
+                PSS_PROCESS_INFORMATION_32 processInformation;
+
+                result = PssQuerySnapshot(snapshotHandle, PSS_QUERY_INFORMATION_CLASS.PSS_QUERY_PROCESS_INFORMATION, &processInformation, Marshal.SizeOf<PSS_PROCESS_INFORMATION_32>());
+
+                if (result != 0)
+                {
+                    throw new Win32Exception(result, $"PssQuerySnapshot with PSS_QUERY_PROCESS_INFORMATION (32 bits) failed with code {result}");
+                }
+
+                userTime = processInformation.UserTime;
+                kernelTime = processInformation.KernelTime;
+                memoryUsage = (long)processInformation.PrivateUsage;
+            }
+
+            userProcessorTime = TimeSpan.FromTicks(userTime);
+            systemCpuTime = TimeSpan.FromTicks(kernelTime);
+            privateMemorySize = memoryUsage;
+            return true;
+        }
+        finally
+        {
+            if (snapshotHandle != IntPtr.Zero)
+            {
+                var result = PssFreeSnapshot(CurrentProcessHandle, snapshotHandle);
+
+                if (result != 0)
+                {
+                    Log.Error<IntPtr, int>("PssFreeSnapshot returned an error, the tracer might be leaking memory. Handle: {Handle}. Error code: {Result}.", snapshotHandle, result);
+                }
+            }
+        }
+    }
+
+    [DllImport("kernel32.dll")]
+    private static extern int PssCaptureSnapshot(IntPtr processHandle, PSS_CAPTURE_FLAGS captureFlags, int threadContextFlags, out IntPtr snapshotHandle);
+
+    [DllImport("kernel32.dll")]
+    private static extern int PssFreeSnapshot(IntPtr processHandle, IntPtr snapshotHandle);
+
+    [DllImport("kernel32.dll")]
+    private static extern unsafe int PssQuerySnapshot(IntPtr snapshotHandle, PSS_QUERY_INFORMATION_CLASS informationClass, void* buffer, int bufferLength);
+
+    [StructLayout(LayoutKind.Sequential)]
+    private struct PSS_THREAD_INFORMATION
+    {
+        public int ThreadsCaptured;
+        public int ContextLength;
+    }
+
+    [StructLayout(LayoutKind.Sequential, Pack = 8)]
+    private unsafe struct PSS_PROCESS_INFORMATION_64
+    {
+        public uint ExitStatus;
+        public IntPtr PebBaseAddress;
+        public nint AffinityMask;
+        public int BasePriority;
+        public uint ProcessId;
+        public uint ParentProcessId;
+        public PSS_PROCESS_FLAGS Flags;
+        public long CreateTime;
+        public long ExitTime;
+        public long KernelTime;
+        public long UserTime;
```

Comment:
Since the [FILETIME structure](https://learn.microsoft.com/en-us/windows/win32/api/minwinbase/ns-minwinbase-filetime) is composed of two DWORDS (which I believe are unsigned 32-bit values), should this be a `ulong` value instead of a `long` value?

Reply:
Ultimately I don't believe it will change anything because we give them to `TimeSpan.FromTicks` which expects a long (so we have to cast them back). Still, I suppose it's better to get the marshalled structure right.

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/5304/commits/0fe295ca7c0793caa050f01d8a9e3211e76e452b